include/dxc/WinAdapter.h

```diff
@@ -912,16 +912,60 @@ unsigned int SysStringLen(const BSTR bstrString);
 #define CP_ACP 0
 #define CP_UTF8 65001 // UTF-8 translation.
 
-// Convert Windows codepage value to locale string
-const char *CPToLocale(uint32_t CodePage);
+// RAII style mechanism for setting/unsetting a locale for the specified Windows
+// codepage
+class ScopedLocale {
+  const char *m_locale;
+
+public:
+  explicit ScopedLocale(uint32_t codePage)
+      : m_locale(SetLocaleForCodePage(LC_ALL, codePage)) {}
```

Comment:
Why do we use `LC_ALL` instead of `LC_CTYPE` to only set the encoding?  Couldn't this potentially fail on a system without `en_US` installed/available?

Reply:
How was this resolved?  Is setting all other locale properties to `en_US` really not a problem?  Why is it necessary?

Reply:
I think the overall assumptions on en-US in the system needs some attention.  I am choosing the path of least resisitance at the moment, but leaving the existing behaviors untouched.  I'll file an Issue for this so we can track as a known thing we need to think about and possibly address.

Reply:
Added Issue #6497 to track a full revisit of en-us assumptions here.